includes/process-purchase.php

```diff
@@ -877,7 +877,7 @@ function edd_purchase_form_validate_cc() {
 	$card_data = edd_get_purchase_cc_info();
 
 	// Validate the card zip
-	if ( ! empty( $card_data['card_zip'] ) ) {
+	if ( ! empty( $card_data['card_zip'] ) && 0.00 !== edd_get_cart_total() ) {
```

Comment:
I'd like to see this changed to `&& ! edd_get_cart_total() > 0` to ensure extensions that filter the cart amount cannot cause this to not work due to the strict type check.